## Importing Libraries and Loading the Dataset

In [2]:
import seaborn as sns

In [3]:
df = sns.load_dataset('titanic')

In [4]:
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

### Checking the presence of Null Values 

In [ ]:
df.isnull()   ## True means the presence of Null Value

survived  pclass    sex    age  sibsp  parch   fare  embarked  class  \
0       False   False  False  False  False  False  False     False  False   
1       False   False  False  False  False  False  False     False  False   
2       False   False  False  False  False  False  False     False  False   
3       False   False  False  False  False  False  False     False  False   
4       False   False  False  False  False  False  False     False  False   
..        ...     ...    ...    ...    ...    ...    ...       ...    ...   
886     False   False  False  False  False  False  False     False  False   
887     False   False  False  False  False  False  False     False  False   
888     False   False  False   True  False  False  False     False  False   
889     False   False  False  False  False  False  False     False  False   
890     False   False  False  False  False  False  False     False  False   

       who  adult_male   deck  embark_town  alive  alone  
0    False       False   True        False  False  False  
1    False       False  False        False  False  False  
2    False       False   True        False  False  False  
3    False       False  False        False  False  False  
4    False       False   True        False  False  False  
..     ...         ...    ...          ...    ...    ...  
886  False       False   True        False  False  False  
887  False       False  False        False  False  False  
888  False       False   True        False  False  False  
889  False       False  False        False  False  False  
890  False       False   True        False  False  False  

[891 rows x 15 columns]

In [ ]:
df.isnull().any()   ## True means that column contains Null Values

survived       False
pclass         False
sex            False
age             True
sibsp          False
parch          False
fare           False
embarked        True
class          False
who            False
adult_male     False
deck            True
embark_town     True
alive          False
alone          False
dtype: bool

In [7]:
df.isnull().sum() ## gives the count of null values in a column


survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

### Deleting Rows or Columns with Missing Values

In [8]:
df.shape 

(891, 15)

There are 891 rows and 15 columns in the dataframs

One way to handle missing values is to delete the rows or columns containing them. However, this can result in significant data loss.

In [9]:
df.dropna().shape

(182, 15)

df.drop() returns a new dataframe in which all the rows having any missing value are deleted. We can observe that initially there were 891 rows and after deleting rows with missing value we are left with 182 rows.... this is a significant loss of data. If a column has a large number of missing values, it may be suitable to delete that column.  From df.isnull().sum() we gotta know that deck is having 688 missing values and total rows are 891... thus it is better to entirely delete that column

In [11]:
newDf = df.dropna(axis=1)

df.dropna(axis = 1) drops all the columns with any missing value and returns a new dataframe.... In df we have 4 Columns having empty values (age, embarked, deck and embark_town). After doing df.dropna(axis=1) all these columns got deleted

In [12]:
newDf.shape

(891, 11)

In [13]:
newDf.head()

survived  pclass     sex  sibsp  parch     fare  class    who  adult_male  \
0         0       3    male      1      0   7.2500  Third    man        True   
1         1       1  female      1      0  71.2833  First  woman       False   
2         1       3  female      0      0   7.9250  Third  woman       False   
3         1       1  female      1      0  53.1000  First  woman       False   
4         0       3    male      0      0   8.0500  Third    man        True   

  alive  alone  
0    no  False  
1   yes  False  
2   yes   True  
3   yes  False  
4    no   True

You can use inplace=True to make the operation permanent....instead of creating newDf